Basic testing setup

Pieter12345 opened the pull request:
While not all tests have been implemented yet, it is useful to replace the broken test setup on master so that other branches can more easily add tests to this basic setup.

Pieter12345 committed:
[WIP] Add AssetShareApp tests

Pieter12345 committed:
Extend contract with test contract for more access

Using this method, making all private functions internal will allow the contract extension to wrap these functions that will then allow the JavaScript test file to call them.
This commit does this with the addOwner() function as example.

babeanu-dorian commented on test/AssetShareAppTestHelper.sol:
```diff
@@ -0,0 +1,10 @@
+
+import "../contracts/AssetShareApp.sol";
+
+contract AssetShareAppHelper is AssetShareApp {
```
call this AssetShareAppTestHelper

Pieter12345 replied:
Renamed.

babeanu-dorian commented on test/AssetShareAppTest.js:
```diff
@@ -0,0 +1,124 @@
+        await app.callAddOwner(ADDRESS_1, 0);
+    });
+    
+    // it('should be incremented by any address', async () => {
```
remove commented code

Pieter12345 replied:
Done.

babeanu-dorian commented on test/AssetShareAppTest.js:
```diff
@@ -0,0 +1,124 @@
+        await app.payOwners({timestamp: currentTimeSec + parseInt(await app.payoutPeriod)});
+        
+        // Assert that the owners have received the payout.
+        // TODO - How to get Ether from address?
```
did you find out how ?

Pieter12345 replied:
I've obtained some information and sent that to @preppas to work on this last week. It can be merged for now, and implemented later. The main purpose of this PR is to add an executable test code framework to the project.

babeanu-dorian reviewed (changes requested):
Please look at and address the comments I've made.

Pieter12345 committed:
Rename AssetShareAppTestHelper to match file name

Pieter12345 committed:
Set initializer to be a user, rather than the organization

This will become a requirement as soon as the application can have multiple contract instances (one per asset).

Pieter12345 committed:
Cleanup AssetShareAppTest.js
